src/components/expression/expression-renderer.js

```diff
@@ -630,7 +629,7 @@ genotet.ExpressionRenderer.prototype.dataLoaded = function() {
 
 /** @inheritDoc */
 genotet.ExpressionRenderer.prototype.dataReady = function() {
-  return this.data.tfaData != null;
+  return this.data.tfa.tfaValues != null;
```

Comment:
I don't know when this was changed to 
```js
return this.data.tfaData != null;
```
but it seems to be a mistake.

dataReady should only return true when the view is ready to be rendered. However having tfa doesn't seem to be a good checking condition for rendering. The user should be able to view expression matrix WITHOUT any tfa. That's why I said tfa loading is quite independent of the matrix loading and you need to reduce their dependencies.